libhwmon/wrapadl.cpp

```diff
@@ -140,6 +160,33 @@ return NULL;
 			if (adapterID == last_adapter) {
 				continue;
 			}
+#if ETH_ETHASHCL
```

Comment:
this should be AFTER the logical to physical loop, not inside of it.
The logical gpus are just an annoyance and we should only use the physical ones. 
(i also had a bug in the beginning, because i thought we get the actual physical gpus, but got logical. so i added the mapping after the first implementation)

So you can do a ``for (unsigned gpuindex =0; gpuindex<adlh->adl_gpucount;gpuindex++) ``  and use ``adlh->devs[adlh->phys_logi_device_id[gpuindex]].iBusNumber``